continuous_integration/gpuci/environment-3.10.yaml

```diff
@@ -14,15 +14,17 @@ dependencies:
 - intake>=0.6.0
 - jsonschema
 - lightgbm
-- maturin>=0.12.8
+- maturin>=0.15,<0.16
 - mlflow
 - mock
 - numpy>=1.21.6
 - pandas>=1.4.0
 - pre-commit
 - prompt_toolkit>=3.0.8
 - psycopg2
-- pyarrow>=6.0.1
+- pyarrow>=6.0.2
+# TODO: remove once fastapi has resolved downstream breakage
+- pydantic<1.10.12
```

Comment:
Might be worth rechecking this now that a newer `fastapi` can be used ( https://github.com/dask-contrib/dask-sql/pull/1205 )

Reply:
Yup, looks like the newer `fastapi` has a similar constraint so should be good to remove this